crates/gpui/src/platform/linux/x11/client.rs

```diff
@@ -1262,7 +1268,9 @@ impl LinuxClient for X11Client {
 
                 state = self.0.borrow_mut();
 
-                if Instant::now() + Duration::from_millis(1) >= next_refresh_needed {
+                if Instant::now() + Duration::from_millis(1) >= next_refresh_needed
+                    && state.last_emptied_runnables.elapsed() < Duration::from_millis(100)
```

Comment:
Minor suggestions:

1. Since `Duration::elapsed` calls `Instant::now`, what do you think about reusing the same `Instant`, and expressing both conditions in terms of `now`?
2. Can we give a name to that 100ms duration? what about something like

```rust
let now = Instant::now();
if now + Duration::from_millis(1) >= next_refresh_needed
    && now - state.last_emptied_runnables < MAX_RUNNABLE_QUEUE_TIME
{
    continue 'run_loop;
}
```

Also, as a side note, do you know what is up with that `Duration::from_millis(1)` term that's added in the already-existing condition?

Reply:
The idea behind the existing condition was to ensure we get round to re-rendering before the timer fires. If we're late, we've missed the refresh rate, but if we're early we get that time back next time.

It may be too much of a margin though, so if we trust that most runnables will be significantly faster than 1ms, we could remove it and get ourselves some more breathing room